File: jscomp/test/name_mangle_test.js
'use strict';

var Mt = require("./mt.js");
var Block = require("../../lib/js/block.js");

var suites = [/* [] */0];

var test_id = [0];

function eq(loc, x, y) {
  test_id[0] = test_id[0] + 1 | 0;
  suites[0] = /* :: */[
    /* tuple */[
      loc + (" id " + String(test_id[0])),
      (function () {
          return /* Eq */Block.__(0, [
                    x,
                    y
                  ]);
        })
    ],
    suites[0]
  ];
  return /* () */0;
}

function f0(x) {
  var old = x.open;
  x.open = old + 1 | 0;
  return x.open;
}

function f1(x) {
  var old = x.in;
  x.in = old + 1 | 0;
  return x.in;
}

function f2(x) {
  var old = x.MAX_LENGTH;
  x.MAX_LENGTH = old + 1 | 0;
  return x.MAX_LENGTH;
}

function f3(x) {
  var old = x.Capital;
  x.Capital = old + 1 | 0;
  return x.Capital;
}

function f4(x) {
  var old = x._open;
  x._open = old + 1 | 0;
  return x._open;
}

function f5(x) {
  var old = x.open;
  x.open = old + 1 | 0;
  return x.open;
}

function f6(x) {
  var old = x["'x"];
  x["'x"] = old + 1 | 0;
  return x["'x"];
}

function f7(x) {
  var old = x._Capital;
  x._Capital = old + 1 | 0;
  return x._Capital;
}

function f8(x) {
  var old = x._MAX;
  x._MAX = old + 1 | 0;
  return x._MAX;
}

function f9(x) {
  var old = x.__;
  x.__ = old + 1 | 0;
  return x.__;
}

function f10(x) {
  var old = x.__x;
  x.__x = old + 1 | 0;
  return x.__x;
}

function f11(x) {
  var old = x._;
  x._ = old + 1 | 0;
  return x._;
}

function f12(x) {
  var old = x.__;
  x.__ = old + 1 | 0;
  return x.__;
}

eq("File \"name_mangle_test.ml\", line 85, characters 7-14", f0(({open:0})), 1);

eq("File \"name_mangle_test.ml\", line 86, characters 7-14", f1(({in:0})), 1);

eq("File \"name_mangle_test.ml\", line 87, characters 7-14", f2(({MAX_LENGTH:0})), 1);

eq("File \"name_mangle_test.ml\", line 88, characters 7-14", f3(({Capital:0})), 1);

eq("File \"name_mangle_test.ml\", line 89, characters 7-14", f4(({_open:0})), 1);

eq("File \"name_mangle_test.ml\", line 90, characters 7-14", f5(({open:0})), 1);

eq("File \"name_mangle_test.ml\", line 91, characters 7-14", f6(({ "'x" :0} )), 1);

eq("File \"name_mangle_test.ml\", line 92, characters 7-14", f7(({_Capital:0})), 1);

eq("File \"name_mangle_test.ml\", line 93, characters 7-14", f8(({_MAX:0})), 1);

eq("File \"name_mangle_test.ml\", line 94, characters 7-14", f9(({__:0})), 1);

eq("File \"name_mangle_test.ml\", line 95, characters 7-14", f10(({__x:0})), 1);

eq("File \"name_mangle_test.ml\", line 96, characters 7-14", f11(({_:0})), 1);

eq("File \"name_mangle_test.ml\", line 97, characters 7-14", f12(({__:0})), 1);

Mt.from_pair_suites("File \"name_mangle_test.ml\", line 101, characters 23-30", suites[0]);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.f0 = f0;
exports.f1 = f1;
exports.f2 = f2;
exports.f3 = f3;
exports.f4 = f4;
exports.f5 = f5;
exports.f6 = f6;
exports.f7 = f7;
exports.f8 = f8;
exports.f9 = f9;
exports.f10 = f10;
exports.f11 = f11;
exports.f12 = f12;
/*  Not a pure module */

File: jscomp/test/name_mangle_test.ml
let suites : Mt.pair_suites ref = ref []
let test_id = ref 0

let eq loc x y =
  incr test_id ;
  suites :=
    (loc ^ " id " ^ string_of_int !test_id, fun _ -> Mt.Eq (x, y)) :: !suites

(* FIXME: paren needed here {[ x##xh#= (g z ) ]} *)
let f0 (x : < _open: int [@bs.set] > Js.t) =
  let old = x##_open in
  x ## _open #= (old + 1) ;
  x##_open

let f1 (x : < _in: int [@bs.set] > Js.t) =
  let old = x##_in in
  x ## _in #= (old + 1) ;
  x##_in

let f2 (x : < _MAX_LENGTH: int [@bs.set] > Js.t) =
  let old = x##_MAX_LENGTH in
  x ## _MAX_LENGTH #= (old + 1) ;
  x##_MAX_LENGTH

let f3 (x : < _Capital: int [@bs.set] > Js.t) =
  let old = x##_Capital in
  x ## _Capital #= (old + 1) ;
  x##_Capital

let f4 (x : < _open__: int [@bs.set] > Js.t) =
  let old = x##_open__ in
  x ## _open__ #= (old + 1) ;
  x##_open__

let f5 (x : < open__: int [@bs.set] > Js.t) =
  let old = x##open__ in
  x ## open__ #= (old + 1) ;
  x##open__

(* < _ : int > -> is a syntax error *)

let f6 (x : < _'x: int [@bs.set] > Js.t) =
  let old = x##_'x in
  x ## _'x #= (old + 1) ;
  x##_'x

let f7 (x : < _Capital__: int [@bs.set] > Js.t) =
  let old = x##_Capital__ in
  x ## _Capital__ #= (old + 1) ;
  x##_Capital__

let f8 (x : < _MAX__: int [@bs.set] > Js.t) =
  let old = x##_MAX__ in
  x ## _MAX__ #= (old + 1) ;
  x##_MAX__

let f9 (x : < __: int [@bs.set] > Js.t) =
  let old = x##__ in
  x ## __ #= (old + 1) ;
  x##__

let f10 (x : < __x: int [@bs.set] > Js.t) =
  let old = x##__x in
  x ## __x #= (old + 1) ;
  x##__x

(* triple _ *)
let f11 (x : < ___: int [@bs.set] > Js.t) =
  let old = x##___ in
  x ## ___ #= (old + 1) ;
  x##___

(* quad _ *)
let f12 (x : < ____: int [@bs.set] > Js.t) =
  let old = x##____ in
  x ## ____ #= (old + 1) ;
  x##____

let () =
  eq __LOC__ (f0 [%raw "{open:0}"]) 1 ;
  eq __LOC__ (f1 [%raw "{in:0}"]) 1 ;
  eq __LOC__ (f2 [%raw "{MAX_LENGTH:0}"]) 1 ;
  eq __LOC__ (f3 [%raw "{Capital:0}"]) 1 ;
  eq __LOC__ (f4 [%raw "{_open:0}"]) 1 ;
  eq __LOC__ (f5 [%raw "{open:0}"]) 1 ;
  eq __LOC__ (f6 [%raw {|{ "'x" :0} |}]) 1 ;
  eq __LOC__ (f7 [%raw "{_Capital:0}"]) 1 ;
  eq __LOC__ (f8 [%raw "{_MAX:0}"]) 1 ;
  eq __LOC__ (f9 [%raw "{__:0}"]) 1 ;
  eq __LOC__ (f10 [%raw "{__x:0}"]) 1 ;
  eq __LOC__ (f11 [%raw "{_:0}"]) 1 ;
  eq __LOC__ (f12 [%raw "{__:0}"]) 1

;;
Mt.from_pair_suites __LOC__ !suites
